tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -10055,17 +10056,11 @@ bool FlattenedTypeIterator::considerLeaf()
   case FlattenedIterKind::FK_Fields:
     if (pushTrackerForType(tracker.CurrentField->getType(), nullptr)) {
       result = considerLeaf();
-    } else {
-      // Pop empty struct.
-      m_typeTrackers.pop_back();
```

Comment:
This would cause an empty struct field to act as an end marker for its containing struct. We should not pop now but rather keep iterating to the next field. This is a fix for #1873